src/Compiler/Optimize/Optimizer.fs

```diff
@@ -3421,6 +3427,17 @@ and TryInlineApplication cenv env finfo (tyargs: TType list, args: Expr list, m)
 
         if isGetHashCode then None else
 
+        let isApplicationComplete =
```

Comment:
What is this about?

Reply:
This is that the lambda being applied is a complete expression.  The name is actually wrong though, it should be named isApplicationPartialExpr.